### ISSUE TITLE
Add NestJS icon support and allow alternation between NestJS and AngularJS

### ISSUE THREAD

**Issue by ryzzaki** [moxer-theme/moxer-icons-code#44]
**Is your feature request related to a problem? Please describe.**
The current Moxer Icons pack for `vscode` contains a set of icons with a support for Angular enabled projects. These icons are however, also propagated into NestJS projects with similar naming conventions such as:

```
foobar.service.ts
foobar.module.ts
foobar.controller.ts
foobar.guard.ts
```

As a result, the typical NestJS project ends up looking like this in Moxer:
<img width="270" alt="Screenshot 2022-11-29 at 14 37 05" src="https://user-images.githubusercontent.com/31356058/204543599-49ed63c7-7240-442b-addf-a6a1d06702cd.png">

As opposed to Material Icon Theme pack which supports NestJS projects:
<img width="270" alt="Screenshot 2022-11-29 at 14 37 51" src="https://user-images.githubusercontent.com/31356058/204543795-bb773e04-93c4-456e-aed8-0e681c374831.png">

**Describe the solution you'd like**
The ideal solution would be to either:

1. Detect `nest-cli.json` or `angular-cli.json` and adjust the icon pack accordingly
2. Allow the developer to manually toggle between icon packs (a solution implemented in `Material Icon Theme`)

Solution implemented in `Material Icon Theme` for the same exact issue:
<img width="270" alt="Screenshot 2022-11-29 at 14 37 51" src="https://user-images.githubusercontent.com/31356058/204544491-cc1accce-dd0b-4822-9919-b47fc9b6ddc3.png">

**Additional context**
It can be unusual for NestJS only developers to use Moxer and then end up getting Angular only icon pack. This could deter many from using this clean extension. Besides, NestJS is a fast growing framework for serverside applications, implementing this feature could be a great incentive to also include the NestJS community.

**Comment by equinusocio** [comment]
The cleaner and safiest way is to rely on language ID's instead of file names/extensions. As long as there are both `angular` and `nest` languages the icon matching would work.

**Comment by ryzzaki** [comment]
@equinusocio not too familiar with how vscode theme pack extensions work on vscode, but would it just involve adding something like:

```
export const languageIds = {
	nim: '_file_nimble',
	...
	nest: '_file_nest',
	...
```

to `languageIds.ts`?

Would you like me to create a PR with these changes? I can source the `nestjs` SVGs for implementation from other MIT licensed extensions, but not sure if that's something you'd rather design yourself? Let me know on how you'd like to proceed with this issue.

**Comment by equinusocio** [comment]
Yes, icons are hand made/tweaked to match the overal UI and use the theme color palette. Are the languages id called `nim` and `nest`?

**Comment by ryzzaki** [comment]
@equinusocio I am not sure what `languageId` it has or if there is even one for the `nest` framework. I cloned your repository and tested out some changes, I don't think relying on language identifier will be a good solution, because the `nestjs` framework shares the same naming convention as `angular`. So when you have:
```
foobar.service.ts
```
Then you have to decide whether to render an `angular` icon or `nestjs` icon. I think the icon pack solution will be better after all.

Because in the `fileExtensions.ts` we can't have duplicate keys for things like `service.ts` or `component.ts` to accommodate for both, `nest` and `angular` at the same time. Would it be possible to add a manual override for the icon pack settings?

**Comment by equinusocio** [comment]
Unfortunately there are not language IDs for `nest` and `angular` forcing us to rely only on file names. But since bot framework use the same naming convention is really hard to determine which framework is used. 

I don't have the knowledge to implement a complex system like the package you mentioned, and even if possible, i'm not sure i want to add such complexity. Probably the simple way is to remove angular icons to such files and show just the language icon (ts,js,ecc) instead of the framework specific.

**Comment by ryzzaki** [comment]
In that case, if you don't want to introduce dynamic rules and similar complexity to Moxer, then I and many other users would very much appreciate it if the theme could maximise consistency over framework adaptability. 

Relying on the final file extensions like `.ts` or `.tsx` is in my opinion better than `.service.ts` as there might be more frameworks with the same naming convention. It ends up looking more consistent in the editor and doesn't distract too much from the mixed batch of random icons from non-relevant frameworks. 

Let me know if you'll be going through with the removal of `angular` icons or if you'd want me to draft a pull request to implement this change. Thanks for continually maintaining this sleek and clean theme!
